### ISSUE TITLE
SmartIR cannot working after update Hassio

### ISSUE THREAD

**Issue by nguyenmnam** [smartHomeHub/SmartIR#783]
SmartIR cannot working after update, I install new version of hassio and new component of Smart IR but it still not working. Some one can help me plz! Many thanks!
***My config:
smartir:
climate:
  - platform: smartir
    name: Daikin 3
    device_code: 1101
    controller_data: remote.t3_remote
  - platform: smartir
    name: Daikin 1
    device_code: 1101
    controller_data: remote.t1_remote

****Notifications:
Invalid config
The following integrations and platforms could not be set up:
smartir.climate
Please check your config and [logs](https://nambihouse.duckdns.org/config/logs).

***LOGS:
Logger: aiohttp.server
Source: /usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py:405
First occurred: 4:42:03 PM (2 occurrences)
Last logged: 4:42:03 PM

Unhandled exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1183, in _sendfile_fallback
    read = await self.run_in_executor(None, file.readinto, view)
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 514, in start
    resp, reset = await task
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 460, in _handle_request
    reset = await self.finish_response(request, resp, start_time)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 613, in finish_response
    await prepare_meth(request)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_fileresponse.py", line 286, in prepare
    return await self._sendfile(request, fobj, offset, count)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_fileresponse.py", line 99, in _sendfile
    await loop.sendfile(transport, fobj, offset, count)
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1162, in sendfile
    return await self._sendfile_fallback(transport, file,
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1192, in _sendfile_fallback
    await proto.restore()
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 263, in restore
    self._transport.resume_reading()
  File "/usr/local/lib/python3.9/asyncio/sslproto.py", line 343, in resume_reading
    self._ssl_protocol._transport.resume_reading()
AttributeError: 'NoneType' object has no attribute 'resume_reading'

**Comment by Tamsy** [comment]
Using SmartIR with a couple of Daikin AC's here too.

Received a message within HA telling me to upgrade SmartIR to v.1.17.5.

- HA core-2022.2.5
- HA supervisor-2022.01.1

Deleted:
`/config/custom_components/smartir`
and replaced it with :
`/SmartIR-1.17.5/custom_components/smartir`
and copied folder:
`/SmartIR-1.17.5/codes` into `/config/custom_components/smartir`
deleted:
`/config/custom_components/smartir/fan` and `/config/custom_components/smartir/media_player`
since I am using the climate functions only.

The folder structure looks like:
```
<config directory>/
|-- custom_components/
|   |-- smartir/
|       |-- __init__.py
|       |-- climate.py
|       |-- fan.py
|       |-- media_player.py
|       |-- etc...
|       |
|       |-- codes/
|           |
|           |-- climate/
|               |-- 1000.json
|               |-- 1020.json
|               |-- 1021.json
|               |-- etc...
```

After restarting HA all is working as expected (incl. SmartIR).

Are you sure you have thoroughly followed the installation instructions ?

**Comment by nguyenmnam** [comment]
I did as you guide but it still not working, and the erro like this!
![11](https://user-images.githubusercontent.com/56904056/153336737-0e2b7636-df5e-4814-9788-15c9e79d71b7.jpg)

**Comment by Tamsy** [comment]
Give your devices a unique_id* within their containers inside configuration.yaml. Additionally I have added a delay*

I.e.:

```
  smartir:

  climate:
    - platform: smartir
      name: Daikin 3
      unique_id: daikin_3
      device_code: 1101
      delay: 0.5
      controller_data: remote.t3_remote

    - platform: smartir
      name: Daikin 1
      unique_id: daikin_1
      device_code: 1101
      delay: 0.5
      controller_data: remote.t1_remote
```
*1: _unique_id (Optional): An ID that uniquely identifies this device. If two devices have the same unique ID, Home Assistant will raise an exception._
*2: _delay (Optional): Adjusts the delay in seconds between multiple commands. The default is 0.5_ 

See: https://github.com/smartHomeHub/SmartIR/blob/master/docs/CLIMATE.md

**Comment by nguyenmnam** [comment]
It still not working :((
![Uploading 2C95C8FF-6536-43EC-8D24-947800762062.png…]()

**Comment by Tamsy** [comment]
> It still not working!

Have you executed a HA-restart after having edited configuration.yaml?

Also please paster again what the logfile is saying about the latter *after* you have restarted HA.

And last but not least: Instead of pasting screenshots or raw log and config snippets, copy the relevant part(s) you want to show into the editor window here, highlight those parts(s) and click the sign <> on top of the edor menu bar to have it display correctly.

**Comment by nguyenmnam** [comment]
> > Nó vẫn không hoạt động!
> 
> Bạn đã thực hiện khởi động lại HA sau khi chỉnh sửa cấu hình.yaml chưa?
> 
> Ngoài ra, vui lòng dán lại những gì tệp log nói về tệp _sau sau khi_ bạn đã khởi động lại HA.
> 
> Và cuối cùng nhưng không kém phần quan trọng: Thay vì dán ảnh chụp màn hình hoặc đoạn nhật ký thô và đoạn mã cấu hình, hãy sao chép (các) phần có liên quan mà bạn muốn hiển thị vào cửa sổ trình chỉnh sửa tại đây, đánh dấu (các) phần đó và nhấp vào dấu <> trong trình đơn trình chỉnh sửa thanh bên trên để hiển thị chính xác.

1. I restart HA after change code in config.yaml
2. I'll show log right now

**Comment by Tamsy** [comment]
> * controller_data: remote.t3_remote

What are these? Do they show up exactly like this as entities in Developer Tools?

However, it seems that you are not dealing with a bug but with a configuration problem. Thus you should post the problem with the Home Assistant forum rather than with the issue tracker. I.e. look [here](https://community.home-assistant.io/t/smartir-control-your-climate-tv-and-fan-devices-via-ir-rf-controllers/100798).

**Comment by vacari** [comment]
I'm having exactly the same problem. I believe other people too:
https://community.home-assistant.io/t/smartir-control-your-climate-tv-and-fan-devices-via-ir-rf-controllers/100798/1123

Now, even going back some previous backups (including restoring the virtual machine backup), and manually updating the component (SmartIR v.1.17.5) the problem persists.
Although I haven't made any configuration changes to SmartIR for years.
It looks like some external dependency. Are there any changed IP/ports that need to be cleared on the firewall?

Sorry if I don't know how to paste the codes correctly, but some information:
core-2022.2.9
supervisor-2022.01.1
Home Assistant OS 7.4 (running on Hyper-V)

Message:
```
Invalid config
The following integrations and platforms could not be set up:

smartir.media_player
smartir.climate
Please check your config and [logs]
```

Full log:
```
2022-02-20 20:12:39 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration smartir which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-02-20 20:12:39 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration alexa_media which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-02-20 20:12:39 WARNING (SyncWorker_2) [homeassistant.loader] We found a custom integration sonoff which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-02-20 20:12:51 WARNING (MainThread) [homeassistant.setup] Setup of smartir is taking over 10 seconds.
2022-02-20 20:13:11 ERROR (MainThread) [homeassistant.components.updater] Timeout fetching Home Assistant update data
2022-02-20 20:13:41 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: smartir
2022-02-20 20:14:42 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: smartir
2022-02-20 20:15:42 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: smartir
2022-02-20 20:16:42 WARNING (MainThread) [homeassistant.bootstrap] Waiting on integrations to complete setup: smartir
2022-02-20 20:17:41 ERROR (MainThread) [homeassistant.setup] Setup of smartir is taking longer than 300 seconds. Startup will proceed without waiting any longer
2022-02-20 20:17:41 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform smartir.media_player: Unable to set up component.
2022-02-20 20:17:41 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform smartir.climate: Unable to set up component.
```

configuration.yaml:

`############################### SMARTIR Integration
smartir:
#  update_branch: rc

############# SMARTIR Climate Integration
climate:
  - platform: smartir
    name: Office AC
    unique_id: office_ac
    device_code: 1101
    controller_data: remote.broadlink_capo_remote
    temperature_sensor: sensor.sonoff_a440002065_temperature
    humidity_sensor: sensor.sonoff_a440002065_humidity

############# SMARTIR Media Player Integration
media_player:
  - platform: smartir
    name: SmartIR Capo TV
    unique_id: smartir_capotv
    device_code: 1060
    controller_data: remote.broadlink_capo_remote
    power_sensor: binary_sensor.ping_capotv`

**Comment by vassilis-panos** [comment]
I also see another error here: 
`20:13:11 ERROR (MainThread) [homeassistant.components.updater] Timeout fetching Home Assistant update data`
Does your VM have internet access? You could try setting up smartir as follows: 

```
smartir:
  check_updates: false
```

**Comment by vacari** [comment]
> Também vejo outro erro aqui: `20:13:11 ERROR (MainThread) [homeassistant.components.updater] Timeout fetching Home Assistant update data` sua VM tem acesso à Internet? Você pode tentar configurar o smartir da seguinte forma:
> 
> ```
> smartir:
>   check_updates: false
> ```

Hey! Thanks for your kind reply!
I had already done that. Then threw error: YAML missing
I had updated (SmartIR v.1.17.5) but the Codes folder did not. (Shouldn't be finding the Source). I think that's the problem.
So I added everything from here:
https://github.com/smartHomeHub/SmartIR/tree/master/codes

Looks like everything is fine now. Do you know the IP where you look for updates?

Thank you again.

**Comment by vassilis-panos** [comment]
Updates are fetched from https://raw.githubusercontent.com/

**Comment by nguyenmnam** [comment]
> > Também vejo outro erro aqui: `20:13:11 ERROR (MainThread) [homeassistant.components.updater] Timeout fetching Home Assistant update data` sua VM tem acesso à Internet? Você pode tentar configurar o smartir da seguinte forma:
> > ```
> > smartir:
> >   check_updates: false
> > ```
> 
> Hey! Thanks for your kind reply! I had already done that. Then threw error: **JSON** missing I had updated (SmartIR v.1.17.5) but the Codes folder did not. (Shouldn't be finding the Source). I think that's the problem. So I added everything from here: https://github.com/smartHomeHub/SmartIR/tree/master/codes
> 
> Looks like everything is fine now. Do you know the IP where you look for updates?
> 
> Thank you again.

COuld you show me how you do it?? My Hassio still not working with SmartIR :( too saddd!
